File: packages/shared/src/types/document-type.ts
/** Mirrors `DocumentType` in `apps/api/src/modules/drivepass/documents/domain/value_objects/document_type.py`. */
export enum DocumentType {
  Padron = "PADRON",
  Soap = "SOAP",
  RevisionTecnica = "REVISION_TECNICA",
  CertificadoGases = "CERTIFICADO_GASES",
  CertificadoHomologacion = "CERTIFICADO_HOMOLOGACION",
  PermisoCirculacion = "PERMISO_CIRCULACION",
  SeguroParticular = "SEGURO_PARTICULAR",
  LicenciaConducir = "LICENCIA_CONDUCIR",
}

/** Only these are accepted by the vehicle document endpoints — mirrors
 * `VEHICLE_DOCUMENT_TYPES` (LICENCIA_CONDUCIR belongs to a user, not a vehicle). */
export const VEHICLE_DOCUMENT_TYPES: DocumentType[] = [
  DocumentType.Padron,
  DocumentType.Soap,
  DocumentType.RevisionTecnica,
  DocumentType.CertificadoGases,
  DocumentType.CertificadoHomologacion,
  DocumentType.PermisoCirculacion,
  DocumentType.SeguroParticular,
];

export const DOCUMENT_TYPE_LABELS: Record<DocumentType, string> = {
  [DocumentType.Padron]: "Padrón",
  [DocumentType.Soap]: "SOAP",
  [DocumentType.RevisionTecnica]: "Revisión Técnica",
  [DocumentType.CertificadoGases]: "Certificado de Gases",
  [DocumentType.CertificadoHomologacion]: "Certificado de Homologación",
  [DocumentType.PermisoCirculacion]: "Permiso de Circulación",
  [DocumentType.SeguroParticular]: "Seguro Particular",
  [DocumentType.LicenciaConducir]: "Licencia de Conducir",
};

File: apps/api/src/modules/drivepass/documents/domain/value_objects/document_type.py
from enum import StrEnum


class DocumentType(StrEnum):
    """Every document type PassHub knows about. `LICENCIA_CONDUCIR` belongs to
    a user, not a vehicle — it's declared here (mirroring how Identity's
    `AuthProvider` declares values with no working flow yet) so the column
    type doesn't need a breaking migration when driver's licenses are
    implemented. `VEHICLE_DOCUMENT_TYPES` is what every vehicle-document
    endpoint actually accepts."""

    PADRON = "PADRON"
    SOAP = "SOAP"
    REVISION_TECNICA = "REVISION_TECNICA"
    CERTIFICADO_GASES = "CERTIFICADO_GASES"
    CERTIFICADO_HOMOLOGACION = "CERTIFICADO_HOMOLOGACION"
    PERMISO_CIRCULACION = "PERMISO_CIRCULACION"
    SEGURO_PARTICULAR = "SEGURO_PARTICULAR"
    LICENCIA_CONDUCIR = "LICENCIA_CONDUCIR"


VEHICLE_DOCUMENT_TYPES: frozenset[DocumentType] = frozenset(
    {
        DocumentType.PADRON,
        DocumentType.SOAP,
        DocumentType.REVISION_TECNICA,
        DocumentType.CERTIFICADO_GASES,
        DocumentType.CERTIFICADO_HOMOLOGACION,
        DocumentType.PERMISO_CIRCULACION,
        DocumentType.SEGURO_PARTICULAR,
    }
)

DOCUMENT_TYPE_LABELS: dict[DocumentType, str] = {
    DocumentType.PADRON: "Padrón",
    DocumentType.SOAP: "SOAP",
    DocumentType.REVISION_TECNICA: "Revisión Técnica",
    DocumentType.CERTIFICADO_GASES: "Certificado de Gases",
    DocumentType.CERTIFICADO_HOMOLOGACION: "Certificado de Homologación",
    DocumentType.PERMISO_CIRCULACION: "Permiso de Circulación",
    DocumentType.SEGURO_PARTICULAR: "Seguro Particular",
    DocumentType.LICENCIA_CONDUCIR: "Licencia de Conducir",
}
